fontforge/tottfgpos.c

```diff
@@ -502,24 +512,34 @@ static void dumpcoveragetable(FILE *gpos,SplineChar **glyphs) {
 	putshort(gpos,2);		/* Coverage format=2 => range list */
 	putshort(gpos,range_cnt);	/* count of ranges */
 	last = -2; start = -2;		/* start is a index in our glyph array, last is ttf_glyph */
-	r = 0;
-	for ( i=0; glyphs[i]!=NULL; ++i ) {
-	    if ( glyphs[i]->ttf_glyph!=last+1 ) {
-		if ( last!=-2 ) {
-		    putshort(gpos,glyphs[start]->ttf_glyph);	/* start glyph ID */
-		    putshort(gpos,last);			/* end glyph ID */
-		    putshort(gpos,start);			/* coverage index of start glyph */
-		    ++r;
+	// start is the index in the glyph array of the starting glyph. last is the ttf_glyph of the ending glyph.
+	r = 0; // r keeps count of the emitted ranges.
+	// We follow the chain of glyphs, ending and emitting a range whenever there is a discontinuity.
+	for (i=0; glyphs[i]!=NULL; i++) {
+		if (i > 0 && glyphs[i]->ttf_glyph <= glyphs[i-1]->ttf_glyph)
+			// LogError(_("Glyphs must be ordered when creating coverage table"));
```

Comment:
Ah, good thing I compiled with `-Wall`:

~~~~
tottfgpos.c: In function 'dumpcoveragetable':
tottfgpos.c:519:6: warning: suggest explicit braces to avoid ambiguous 'else' [-Wparentheses]
   if (i > 0 && glyphs[i]->ttf_glyph <= glyphs[i-1]->ttf_glyph)
~~~~

Because you commented out the line, this if statement is actually falling though to the next if statement e.g.

if (A)
    if (B)